From: Aditya Vaidya <kroq.gar78@gmail.com>
Date: Sat, 11 Apr 2015 02:16:53 -0500
Subject: remove-installer

---
 src/gui/game/GameController.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/game/GameController.cpp b/src/gui/game/GameController.cpp
index 43f0396..e6f20fc 100644
--- a/src/gui/game/GameController.cpp
+++ b/src/gui/game/GameController.cpp
@@ -731,11 +731,11 @@ void GameController::Tick()
 		((LuaScriptInterface*)commandInterface)->Init();
 #endif
 #if !defined(MACOSX) && !defined(NO_INSTALL_CHECK)
-		if(!Client::Ref().GetPrefBool("InstallCheck", false))
+		/*if(!Client::Ref().GetPrefBool("InstallCheck", false))
 		{
 			Client::Ref().SetPref("InstallCheck", true);
 			Install();
-		}
+		}*/
 #endif
 		firstTick = false;
 	}
