Description: Fix compiling on the Raspberry Pi
Origin: http://powdertoy.co.uk/Discussions/Thread/View.html?Thread=14606&PageNum=0
diff --git a/Makefile b/Makefile
index b2d4570..dddd2db 100644
--- a/Makefile
+++ b/Makefile
@@ -7,9 +7,9 @@ OFLAGS := -O3 -ffast-math -ftree-vectorize -funsafe-math-optimizations
 LFLAGS := -lpthread -lSDL -lfftw3f -lm -lbz2 -lX11 -llua5.1 -lrt
 LFLAGS_X := -lm -lbz2 -lSDLmain
 LFLAGS_WIN := -lmingw32 -lgnurx -lws2_32 -lSDLmain -lpthread -lSDL -lfftw3f -lm -lbz2 -llua5.1
-MFLAGS_SSE3 := -march=native -DX86 -DX86_SSE3 -msse3
-MFLAGS_SSE2 := -march=native -DX86 -DX86_SSE2 -msse2
-MFLAGS_SSE := -march=native -DX86 -DX86_SSE
+MFLAGS_SSE3 := -march=armv5 -DX86 -DX86_SSE3
+MFLAGS_SSE2 := -march=armv5 -DX86 -DX86_SSE2
+MFLAGS_SSE := -march=armv5 -DX86 -DX86_SSE
 FLAGS_DBUG := -Wall -pg -g
 COMPILER := gcc
 LINUX_TARG := powder-64-sse2 powder-sse powder-sse2
@@ -35,17 +35,17 @@ powder-sse.exe: build/powder-sse.exe
 
 # general compiler flags
 build/powder: CFLAGS += -DINTERNAL -DLIN64 $(OFLAGS)
-build/powder-debug: CFLAGS += -m32 -DLIN32 $(FLAGS_DBUG)
-build/powder-sse3 build/powder-sse2 build/powder-sse build/powder-sse3-opengl: CFLAGS += -m32 -DLIN32 $(OFLAGS)
-build/powder-64-sse3 build/powder-64-sse2 build/powder-64-sse3-opengl: CFLAGS += -m64 -DLIN64 $(OFLAGS)
-build/powder-debug-64: CFLAGS += -m64 -DLIN64 $(FLAGS_DBUG)
+build/powder-debug: CFLAGS += -mabi=aapcs-linux -DLIN32 $(FLAGS_DBUG)
+build/powder-sse3 build/powder-sse2 build/powder-sse build/powder-sse3-opengl: CFLAGS += -mabi=aapcs-linux -DLIN32 $(OFLAGS)
+build/powder-64-sse3 build/powder-64-sse2 build/powder-64-sse3-opengl: CFLAGS += -mabi=aapcs-linux -DLIN64 $(OFLAGS)
+build/powder-debug-64: CFLAGS += -mabi=aapcs-linux -DLIN64 $(FLAGS_DBUG)
 build/powder-sse3.exe build/powder-sse2.exe build/powder-sse.exe: CFLAGS += -mwindows -DWIN32 $(OFLAGS)
 build/powder-64-sse3-opengl build/powder-sse3-opengl: CFLAGS += -DOGLR -DPIX32OGL -DPIXALPHA
 
 # SSE flags:
-build/powder build/powder-sse3 build/powder-sse3-opengl build/powder-64-sse3 build/powder-64-sse3-opengl build/powder-debug build/powder-debug-64 build/powder-sse3.exe: CFLAGS += -march=native -DX86 -DX86_SSE3 -msse3
-build/powder-sse2 build/powder-64-sse2 build/powder-sse2.exe: CFLAGS += -march=native -DX86 -DX86_SSE2 -msse2
-build/powder-sse build/powder-sse.exe: CFLAGS += -march=native -DX86 -DX86_SSE
+build/powder build/powder-sse3 build/powder-sse3-opengl build/powder-64-sse3 build/powder-64-sse3-opengl build/powder-debug build/powder-debug-64 build/powder-sse3.exe: CFLAGS += -march=armv5 -DX86 -DX86_SSE3
+build/powder-sse2 build/powder-64-sse2 build/powder-sse2.exe: CFLAGS += -march=armv5 -DX86 -DX86_SSE2
+build/powder-sse build/powder-sse.exe: CFLAGS += -march=armv5 -DX86 -DX86_SSE
 
 # libs:
 build/powder build/powder-debug build/powder-sse3 build/powder-sse2 build/powder-sse build/powder-sse3-opengl build/powder-debug-64 build/powder-64-sse3 build/powder-64-sse2 build/powder-64-sse3-opengl: LIBS += $(LFLAGS)
diff --git a/src/misc.c b/src/misc.c
index 322a56a..61a48ca 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -532,30 +532,7 @@ void *file_load(char *fn, int *size)
 
 int cpu_check(void)
 {
-#ifdef MACOSX
-	return 0;
-#else
-#ifdef X86
-	unsigned af,bf,cf,df;
-	x86_cpuid(0, af, bf, cf, df);
-	if (bf==0x68747541 && cf==0x444D4163 && df==0x69746E65)
-		amd = 1;
-	x86_cpuid(1, af, bf, cf, df);
-#ifdef X86_SSE
-	if (!(df&(1<<25)))
-		return 1;
-#endif
-#ifdef X86_SSE2
-	if (!(df&(1<<26)))
-		return 1;
-#endif
-#ifdef X86_SSE3
-	if (!(cf&1))
-		return 1;
-#endif
-#endif
-#endif
-	return 0;
+return 0;
 }
 
 matrix2d m2d_multiply_m2d(matrix2d m1, matrix2d m2)
