--- a/src/client/Client.cpp
+++ b/src/client/Client.cpp
@@ -582,7 +582,7 @@
 				notifyMessageOfTheDay();
 
 				//Check for updates
-				json::Object versions = objDocument["Updates"];
+				/*json::Object versions = objDocument["Updates"];
 				
 				json::Object stableVersion = versions["Stable"];
 				json::Object betaVersion = versions["Beta"];
@@ -626,7 +626,7 @@
 				if(updateAvailable)
 				{
 					notifyUpdateAvailable();
-				}
+				}*/
 			}
 			catch (json::Exception &e)
 			{
