Description: Update upstream's changelog in README
Author: Aditya Vaidya <kroq.gar78@gmail.com>
Forwarded: no
diff --git a/README b/README
index 3b6aeed..44e6249 100644
--- a/README
+++ b/README
@@ -103,6 +103,133 @@ History:
 
 ---------------------------------------------------------------------------
 
+* Version 81.6 Build 189
+      o Fixed zoom in deco mode
+
+* Version 81.5 Build 188
+      o Add console show button to quick menu
+      o Some under-the-hood tweaks
+
+* Version 81.4 Build 187
+      o Fixed pasting text within existing text
+      o Disable decorations by default
+      o Some under-the-hood tweaks
+
+* Version 81.3 Build 186
+      o Now stickmen can shoot plasma.
+
+* Version 81.3 Build 185
+      o SALT only melts ICE that is above SLTW freezing temperature
+      o Brush size and mouse wheel added to lua
+      o Fix PLNT lag
+      o Some under-the-hood fixes and optimization
+
+* Version 81.2 Build 184
+      o Ensure an array terminator is only written if the signs array actually exists
+
+* Version 81.1 Build 183
+      o Allow INST to be sparked directly with the brush
+      o Lightning direction affected by Newtonian gravity
+      o Reduce rate of lightning creation by clone
+      o Make VOID and BHOL eat STKM again
+      o Prevent STKM falling through activated PVOD with a ctype that isn't STKM
+      o Prevent ACID from eating BGLA
+      o Stop STKM being able to jump on the bottom of the screen
+      o Ensure any clicks on stacked save link sigs only open the first found sign
+      o C5 doesn't explode when touching cold INSL
+      o Some under-the-hood changes
+
+* Version 81.0 Build 182
+      o Minor improvement to text wrapping
+      o Fix STKM causing stacking and falling through some powders
+      o Prevent stacking from ELEC+NEUT, ELEC+water, and CO2 from BUBW
+      o Make ELEC+water form H2 and O2 in the correct proportions
+      o New element: EMBR - replaces all explosion sparks
+      o Fix minor bug with deco sample tool
+      o Gravity modes and Newtonian gravity for fireworks
+      o Prevent STKM floating in gases
+      o Allow STKM head to go through everything
+      o Fix compression during fusion
+      o Only create one NBLE particle for each fusing H2 particle
+      o Completely remove SING formation from fusion
+      o Less advection for FIRW, and don't go faster when multiple fire particles are touching it
+      o Some under-the-hood changes
+
+* Version 80.4 Build 179
+      o Prevent solid particles from stacking in E-Hole
+      o Create BHOL when particles stacked excessively
+      o Avoid stacking in INVS and FILT; collisions now work there so stacking doesn't happen
+      o Fix pressurised INVS graphics
+      o STOR doesn't conduct heat
+      o Ambient heat convection works a little better
+      o EXOT is permeable by NEUT and ELEC
+      o FRZZ behaves more like SNOW
+      o Some under-the-hood changes
+
+* Version 80.3 Build 178
+      o Fix session check on startup
+
+* Version 80.3 Build 177
+      o TTAN only blocks air when it is connected
+      o Some optimization
+
+* Version 80.2 Build 176
+      o Fix under-the-hood return problem
+
+* Version 80.1 Build 175
+      o EXOT mimics Heat View's color scale, and colors appear to cycle
+
+* Version 80.1 Build 174
+      o Nothing new?
+
+* Version 80.0 Beta Build 173
+      o Added EXOT element
+      o WARP is invisible except in Blob mode
+
+* Version 79.7 Build 172
+      o Fix bug with TRON not dying
+
+* Version 79.6 Beta Build 171
+      o Fix the bug with not-spawning fighter
+
+* Version 79.5 Beta Build 170
+      o Deferred loading of save files
+      o Moved soap and stick men graphics things to special functions.
+      o Minor under-the-hood changes
+
+* Version 79.4 Beta Build 169
+      o No rounding errors with triangle brush
+      o Fix SLTW boiling temperature typo
+      o Don't show disabled elements in element search
+      o Soap bubbles are easily formed now. Also they inherit soap's decoration color.
+      o Download progress bar for save opening
+      o Fix bug that causes ptsave URL parsing to fail if there is no '#'
+
+* Version 79.3 Beta Build 168
+      o Add DISPLAY_EFFE back
+      o Element search interface by pressing 'e'
+      o VOID & PVOD can be set to only eat/not eat certain things
+      o Energy type particles automatically go into photons, not pmap
+      o Finish custom lua graphics functions
+      o Disable another way to create invalid elements
+      o Fix crash, spark reset set type without checking if it was a valid number.
+      o Fix lua console compiling with visual studio
+      o Fix lua console freeze/bugs
+      o Fix alt select being off by one every time
+      o Turn on deco before entering the deco editor (not after)
+      o Limit amount of ELEC produced to prevent infinite reactions
+      o Console doesn't enable hud
+      o OXYG fusion
+      o Fix small bugs
+      o Prevent accidental infinite loops in lua
+      o Fusion changes
+      o Register ptsave protocol in Windows installation
+      o Fix the bug with stickman standing on the screen edge
+      o Fix the bug where gel didn't transfer water from sponge
+      o Update gravity mask when loading OPS saves
+      o Fix infinite loop when trying to detach soap with extra ctype bits set
+      o Change order of bomb loops
+
 * Version 79.2 Beta Build 167
       o Fix minor spelling errors
       o Fix crash when using flood-fill or a lot of INST
